lib/mr_bump/slack.rb

```diff
@@ -16,31 +21,85 @@ def bump(version, changes)
       options = {}
       options[:icon_url] = @icon if @icon
       options[:attachments] = [attatchment(version, changes)]
-      notifier.ping ' ', options
+      notifier.ping 'There is a new version bump!', options
+    end
+
+    def notifier
+      @notifier ||= ::Slack::Notifier.new(@webhook, username: @username)
     end
 
     def user
-      [
-        ENV["USER"],
-        `whoami`
-      ].detect { |u| !u.nil? }
+      `git config user.name`.gsub(/\n/, '')
+    end
+
+    def git_url
+      @git_url ||= `git remote get-url origin`
+    end
+
+    def git_user
+      git_url[/[^@]+/].partition('//').last
+    end
+
+    def git_icon
+      'https://avatars.githubusercontent.com/' + git_user
+    end
+
+    def git_link
+      'https://github.com/' + git_user
     end
 
     def repo_name
       `git config --get remote.origin.url`[/[\w\.]+(?=\.git$)/]
     end
 
-    def notifier
-      @notifier ||= ::Slack::Notifier.new(@webhook, username: @username)
+    def jira_ids(changes)
+      changes.split('* ').map { |i| i.split(' - ',3)[1]}.compact
+    end
+
+    def jira_urls(changes)
+      jira_ids(changes).map do |ticket|
+        if ticket!= 'UNKNOWN'
+          '<' + @jira_url + '/browse/' + ticket + '|Jira link - ' + ticket + ">\n"
+        end
+      end.join
     end
 
+    def jira_field(changes)
+      if @jira_url != nil
+        {value: jira_urls(changes), short: true, unfurl_links: false}
+      end
+    end
+
+    def get_sha(id)
+      `git log --grep=#{id} --merges --format=format:%H`.split("\n").first
```

Comment:
In some cases, there are multiple commits with one JIRA ID, so I just assume the most recent is the one we are looking for. Maybe there is a better way of identifying our commit? 90% of the time there is just one.